### PR TITLE
API 11224 Observation lastUpdated Migration

### PR DESCRIPTION
**Pull request by that-guy-is-it**
First PR for the multi-PR task, 11224. This PR was intentionally delayed in the hopes of getting the second PR (for DQ) ready also, but since it is taking longer than expected, I am opening this first PR alone for now.

### CODE REVIEW

**Comment by that-guy-is-it** [comment]
@nmoisan @jtaylor-va @haidong I realized I forgot to ping! Apologies! This is a schema change so you guys need to double check too.

**Comment by nmoisan** [comment]
LGTM, matches our naming convention and datatype. 👍

File: db/migration/V22_0__alter_observation.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE [App].[Observation] ADD [LastUpdated] [DATE] NULL
```
Comment (ilaflamme-va) [inline db/migration/V22_0__alter_observation.sql:1]: Can this be `not null`? 
Is `LastUpdated` populated with the creation date when it is first loaded. Every subsequent change would alter this value.

Comment (ilaflamme-va) [reply db/migration/V22_0__alter_observation.sql:1]: If this doesn't matter, it looks fine to me.

Comment (nmoisan) [reply db/migration/V22_0__alter_observation.sql:1]: you are right actually! I missed that. This matches our design too. We define `LastUpdated` as non-nullable with a default upon first load.

Comment (ilaflamme-va) [reply db/migration/V22_0__alter_observation.sql:1]: Thats what I thought :) Thanks for clarifying.

**Review by ilaflamme-va — CHANGES_REQUESTED** [review]
Make last updated not null.

**Comment by ilaflamme-va** [comment]
A couple things need to happen:
- The ObservationEntity does not have a LastUpdated field, DQ will need an updated and the dependency updated
-  MitreMinimartMaker will need to populate the synthetic entities with a value upon loading of the db. see here: https://github.com/department-of-veterans-affairs/health-apis-datamart-synthetic-records/blob/ebb215eb4cf3a169ffca00181777c37d81c1d679/minimart-manager/src/test/java/gov/va/api/health/minimartmanager/minimart/MitreMinimartMaker.java#L113

**Comment by that-guy-is-it** [comment]
Wouldn't merging this as `null` for the time being be beneficial, since the Observation Entity has to be updated and requires a built synthetic database? A second migration can set it to `not null` once it can be populated.

**Comment by ilaflamme-va** [comment]
A follow-up PR is fine, if this null version is just place-holder as things get sorted out.

**Comment by that-guy-is-it** [comment]
> A follow-up PR is fine, if this null version is just place-holder as things get sorted out.

It is, since a second PR is likely necessary for populating anyway.

**Comment by that-guy-is-it** [comment]
Due to a necessity to add `lastUpdated` to the Datamart Payload separately, the populating of this column will be done in a future PR, as originally planned. For the time being, this migration will merge using a `NULL`, and after the DQ model is updated and the population is done, a second migration will set it to `NOT NULL`.

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Review by ilaflamme-va — APPROVED** [review]

**Pull request merged**